config.go

```diff
@@ -123,8 +133,8 @@ type GCConfig struct {
 
 // LoggingConfig is the tasks logging configuration
 type LoggingConfig struct {
-	Driver  string             `codec:"driver"`
-	Options hclutils.MapStrStr `codec:"options"`
+	Driver  string            `codec:"driver"`
+	Options map[string]string `codec:"options"`
```

Comment:
I needed to switch this to a map from `MaptStrStr` because otherwise message pack would fail in `setConfig`:
```
    2023-09-29T12:25:52.047+0200 [ERROR] agent: error starting agent:
  error=
  | failed to create plugin loader: failed to initialize plugin loader: parsing plugin configurations failed: 1 error occurred:
  | \t* plugin "podman" (driver): failed to dispense plugin: setting config for plugin "podman" (driver) failed: rpc error: code = Unknown desc = SetConfig failed: msgpack decode error [pos 117]: invalid byte descriptor for decoding bytes, got: 0x74
  | 
```

not exactly sure why this happens :/

Reply:
Hum...I'm not sure why this would happen either. But I'm a bit worried about backwards compatibility of changing this value type as msgpack can be a bit fuzzy with types.

I'm als worried about the godoc for `MapStrInt`. It seems like this struct was created to support an unconventional syntax that was used in older version of Nomad: https://github.com/hashicorp/nomad/commit/afb0c5ac452ba3bcf5c12907351dccb4edbc368b.

It's also interesting that the PR mentions that external plugins should not use this, but we don't document this anywhere. First time I hear about it 🙃 

> This is only required for built-in types that have backward
> compatibility constraints.  External drivers should use `BlockAttrs`
> instead, as they see fit.

But, since this is already like this, modifying it would be a breaking change.

Instead, I think it may be better to create a new struct just like this but for driver config instead of task. It's kind of annoying that both need to be kept in-sync (and we should note this in their godoc string), but that's already kind the case since `taskConfigSpec` and `configSpec` would need to change as well.